File: Matrike/matrix.c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>

#define DIM 500

void mnozenjeMatrik(int A[][DIM], int C[][DIM]) { 
    
    for (int i = 0; i < DIM; i++) {
        for (int j = 0; j < DIM; j++) {
            for (int k = 0; k < DIM; k++) {
                C[i][j] += A[i][k] * A[k][j];
            }
        }
    }
}

void readNumbers(char* filename, int arr[][DIM]) {
    FILE* file = fopen(filename, "r");

    for (int i = 0; i < DIM; i++) {
        for (int j = 0; j < DIM; j++) {
            fscanf(file, "%d", &arr[i][j]);
        }
    }

    fclose(file);
}

int main() {
    int A[DIM][DIM],C[DIM][DIM];
    readNumbers("numbers.txt", A);
    
	clock_t start, end;
    double cpu_time_used;
    start = clock(); 
	mnozenjeMatrik(A,C);
    end = clock(); 
    cpu_time_used = ((double) (end - start) * 1000.0) / CLOCKS_PER_SEC;
    printf("Time taken: %f miliseconds\n", cpu_time_used);
	printf("%d\n",C[0][0]);
    return 0;
}

File: Matrike/matrix.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"time"
)

const DIM = 500

func readNumbers(filename string, arr* [500][500]int64) {
	file, _ := os.Open(filename)

	defer file.Close()

	scanner := bufio.NewScanner(file)
	for i := 0; i < DIM; i++ {
		for j := 0; j < 500 && scanner.Scan(); j++ {
			num, err := strconv.ParseInt(scanner.Text(), 10, 64)
			if err != nil {
				continue
			}
			arr[i][j] = num
		}
	}
}


func mnozenjeMatrik(A *[500][500]int64, C *[500][500]int64) {
    for i := 0; i < DIM; i++ {
        for j := 0; j < DIM; j++ {
            for k := 0; k < DIM; k++ {
                C[i][j] += A[i][k] * A[k][j]
            }
        }
    }
}

func main() {
    var A[DIM][DIM]int64
	var C[DIM][DIM]int64
	readNumbers("numbers.txt",&A)
    startTime := time.Now()
	mnozenjeMatrik(&A,&C)
	endTime := time.Now()
    timeTaken := endTime.Sub(startTime).Milliseconds()
    fmt.Printf("Time taken: %d milliseconds\n", timeTaken)
	fmt.Printf("Number: %d\n",C[0][0])
}
